server/app/services/UserService.ts

```diff
@@ -7,6 +7,9 @@ import bcrypt from 'bcryptjs';
 import { IConfig } from 'config';
 import validator from 'validator';
 import { BadRequestError, NotApprovedError, NotFoundError } from '../errors';
+import { generateToken } from '../middlewares/authentication';
+
+const requests: any[] = [];
```

Comment:
Can you define an interface in the UserModel file to give this array a type?

Reply:
And you could also use the [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) data structure to make looking up easier than looping and checking on each iteration.

Reply:
changed to map